tensorrt_llm/mapping.py

```
@@ -224,6 +225,9 @@ def __init__(
         self.auto_parallel = auto_parallel
         self.world_size = world_size
         self.enable_attention_dp = enable_attention_dp
+        if enable_lm_head_tp_in_adp:
+            assert enable_attention_dp, "enable_lm_head_tp_in_adp requires enable_attention_dp"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)